### PR TITLE
add timeout to http client

### PR DESCRIPTION
**Pull request by huguesBouvier**
Fixes issue id 10438212
With env var httpClientTimeoutSecs:
image

Creating an image that takes more than 100s
image

Without env var httpClientTimeoutSecs, creating an image that takes more than 100s
image

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Constants.cs [hunk 1]
```diff
@@ -105,6 +105,8 @@ public static class Constants
 
         public const string WorkloadApiVersion = "2020-10-10";
 
+        public const string HttpClientTimeoutSecs = "httpClientTimeoutSecs";
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Constants.cs:108]: We should call it something like WorkloadApiTimeoutSecs

**Comment by varunpuranik** [comment]
How does EdgeAgent call timing out cause docker pull to get interrupted? Shouldn't that still continue in the background, and finish eventually? At that point, the module creation should succeed?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Constants.cs (hunk 1 above)
Comment (huguesBouvier) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/Constants.cs:108]: I agree, (though it is managementAPI)

**Comment by vadim-kovalyov** [comment]
Is this still needed?

**Pull request closed without merging**